buildrunner/__init__.py

```diff
@@ -250,6 +251,40 @@ def _re_split_filter(text, pattern, maxsplit=0, flags=0):
         '''
         return re.split(pattern, text, maxsplit=maxsplit, flags=flags)
 
+    @staticmethod
+    def _reorder_dependency_steps(config):
+        """
+        Reorders the steps based on the dependencies that are outlined in the config
+        """
+        # Defines configuration keywords, should add to a config validation class
+        keyword_version = 'version'
+        keyword_steps = 'steps'
+        keyword_depends =   'depends'
+        supported_version = 2.0
+
+        if keyword_version not in config.keys() \
+                or config[keyword_version] < supported_version:
+            return config
```

Comment:
To go along with my comment above, it feels like using `depends` without the `version` specified should result in an error as well, right?

Reply:
I added this check and message if there is a step name typo in in the step dependencies. (ie a step in the list which isn't defined in the config)
KeyError: "Not defined 'step3e' listed as a step dependency in the configuration. In the configuration please correct the typo or define step 'step3e'."

Not sure if I should create a new exception type or use a different builtin exception.